include/cuco/detail/hash_functions/murmurhash3.cuh

```diff
@@ -206,4 +209,132 @@ struct MurmurHash3_32 {
   MurmurHash3_fmix32<std::uint32_t> fmix32_;
   std::uint32_t seed_;
 };
+
+template <typename Key>
+struct MurmurHash3_128 {
+  using argument_type = Key;  ///< The type of the values taken as argument
+  using result_type = cuda::std::array<std::uint64_t, 2>;  ///< The type of the hash values produced
+
+  /**
+   * @brief Constructs a MurmurHash3_128 hash function with the given `seed`.
+   *
+   * @param seed A custom number to randomize the resulting hash value
+   */
+  __host__ __device__ constexpr MurmurHash3_128(std::uint64_t seed = 0) : fmix64_{0}, seed_{seed} {}
+
+  /**
+   * @brief Returns a hash value for its argument, as a value of type `result_type`.
+   *
+   * @param key The input argument to hash
+   * @return The resulting hash value for `key`
+   */
+  constexpr result_type __host__ __device__ operator()(Key const& key) const noexcept
+  {
+    return compute_hash(reinterpret_cast<std::byte const*>(&key),
+                        cuco::extent<std::size_t, sizeof(Key)>{});
+  }
+
+  /**
+   * @brief Returns a hash value for its argument, as a value of type `result_type`.
+   *
+   * @tparam Extent The extent type
+   *
+   * @param bytes The input argument to hash
+   * @param size The extent of the data in bytes
+   * @return The resulting hash value
+   */
+  template <typename Extent>
+  constexpr result_type __host__ __device__ compute_hash(std::byte const* bytes,
+                                                         Extent size) const noexcept
+  {
+    uint32_t const BLOCK_SIZE = 16;
```

Comment:
```suggestion
    uint32_t constexpr block_size = 16;
```
nit: build time constant allows more aggressive compiler optimization, cuco also follows cudf guideline for naming thus snake case is preferred in this situation https://github.com/rapidsai/cudf/blob/branch-24.08/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#code-and-documentation-style-and-formatting